Pick up the CLR_RANLIB variable properly, even when using Clang as a compiler

CMAKE_RANLIB is used regardless of whether Clang or GCC is being used

--- a/eng/native/configuretools.cmake
+++ b/eng/native/configuretools.cmake
@@ -41,10 +41,9 @@ if(NOT WIN32 AND NOT CLR_CMAKE_TARGET_BR
 
   locate_toolchain_exec(ar CMAKE_AR)
   locate_toolchain_exec(nm CMAKE_NM)
+  locate_toolchain_exec(ranlib CMAKE_RANLIB)
 
-  if(CMAKE_C_COMPILER_ID MATCHES "GNU")
-    locate_toolchain_exec(ranlib CMAKE_RANLIB)
-  else()
+  if(NOT CMAKE_C_COMPILER_ID MATCHES "GNU")
     locate_toolchain_exec(link CMAKE_LINKER)
   endif()
 
